File: app/src/main/java/ru/samtakoy/core/services/import_utils/ImportCardsException.java
package ru.samtakoy.core.services.import_utils;

public class ImportCardsException extends Exception{


    public static final int ERR_PACK_ID_MISSING = 1;
    public static final int ERR_PACK_WITH_ID_CREATION_NOT_ALLOWED = 2;
    public static final int ERR_WRONG_CARD_PACK = 3;

    private int mErrorId;

    public ImportCardsException(int errorId, String message){
        super(message);

        mErrorId = errorId;
    }

    public int getErrorId(){
        return mErrorId;
    }

}

File: module/services/import/api/src/commonMain/kotlin/ru/samtakoy/importcards/domain/batch/utils/ImportCardsException.kt
package ru.samtakoy.importcards.domain.batch.utils

class ImportCardsException(
    val errorId: Int,
    message: String?
) : Exception(message) {
    companion object {
        const val ERR_PACK_ID_MISSING: Int = 1
        const val ERR_PACK_WITH_ID_CREATION_NOT_ALLOWED: Int = 2
        const val ERR_WRONG_CARD_PACK: Int = 3
    }
}